src/test/java/com/wikia/webdriver/common/core/api/ApiCall.java

```diff
@@ -0,0 +1,85 @@
+package com.wikia.webdriver.common.core.api;
+
+import com.wikia.webdriver.common.core.Helios;
+import com.wikia.webdriver.common.core.annotations.User;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.openqa.selenium.WebDriverException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+/**
+ * Created by wikia on 2015-08-27.
+ */
+public abstract class ApiCall {
+
+  private static String ERROR_MESSAGE = "Problem with API call";
+
+  protected static String URL_STRING = null;
+
+  protected ApiCall() {
+
+  }
+
+
+  abstract protected String getURL();
+
+  /**
+   * Return null if API call doesn't require to be logged in as specific user
+   *
+   * @return User to be logged in while executing API call
+   */
+  abstract protected User getUser();
+
+  /**
+   * Return null when no params should be added to API call
+   *
+   * @return params
+   */
+  abstract protected ArrayList<BasicNameValuePair> getParams();
+
+  public void call() {
+    try {
+      URL url = new URL(getURL());
+      CloseableHttpClient httpClient = HttpClients.createDefault();
+      HttpPost httpPost = getHtppPost(url);
+      // set header
+      if (getUser() != null) {
+        httpPost.addHeader("Authorization", "Bearer " + Helios.getAccessToken(getUser()));
+      }
+      // set query params
+      if (getParams() != null) {
+        httpPost.setEntity(new UrlEncodedFormEntity(getParams(), StandardCharsets.UTF_8));
+      }
+
+      httpClient.execute(httpPost);
+
+      PageObjectLogging.log("CONTENT PUSH", "Content posted to: " + URL_STRING, true);
+    } catch (ClientProtocolException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)